packages/terra-popup/src/_PopupContent.jsx

```diff
@@ -0,0 +1,289 @@
+const KEYCODES = {
+  ESCAPE: 27,
+};
+
+/**
+ * Valid positions for the arrow, applies a margin to the corresponding side.
+ */
+const ARROW_POSITIONS = [
+  'top',
+  'bottom',
+  'left',
+  'right',
+];
+
+/**
+ * Margin value used for calculations.
+ */
+const POPUP_MARGIN = 10;
+
+/**
+ * Rounded corner size to be used when calculating the arrow offset.
+ */
+const CORNER_SIZE = 3;
+
+const propTypes = {
+  /**
+   * The children to be presented as the popup's content.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * The height value in px, to be applied to the content container.
+   */
+  contentHeight: PropTypes.number.isRequired,
+  /**
+   * The width value in px, to be applied to the content container.
+   */
+  contentWidth: PropTypes.number.isRequired,
+  /**
+   * The function that should be triggered when a close is indicated.
+   */
+  onRequestClose: PropTypes.func.isRequired,
+  /**
+   * The arrow to be placed within the popup frame.
+   */
+  arrow: PropTypes.element,
+  /**
+   * The initial arrow position.
+   */
+  arrowPosition: PropTypes.oneOf(ARROW_POSITIONS),
+  /**
+   * CSS classnames that are appended to the popup content body.
+   */
+  classNameInner: PropTypes.string,
+  /**
+   * Whether or not the using the escape key should trigger the onRequestClose callback.
+   */
+  closeOnEsc: PropTypes.bool,
+  /**
+   * Whether or not clicking outside the popup should trigger the onRequestClose callback.
+   */
+  closeOnOutsideClick: PropTypes.bool,
+  /**
+   * Whether or not resizing the screen should trigger the onRequestClose callback.
+   */
+  closeOnResize: PropTypes.bool,
+  /**
+   * The maximum height value in px, to be applied to the content container. Used with responsive behavior for actual height.
+   */
+  contentHeightMax: PropTypes.number,
+  /**
+   * The maximum width value in px, to be applied to the content container. Used with responsive behavior for actual width.
+   */
+  contentWidthMax: PropTypes.number,
+  /**
+   * Should the default behavior, that inserts a header when constraints are breached, be disabled.
+   */
+  isHeaderDisabled: PropTypes.bool,
+  /**
+   * The function returning the frame html reference.
+   */
+  refCallback: PropTypes.func,
+};
+
+const defaultProps = {
+  arrowPosition: 'top',
+  classNameInner: null,
+  closeOnEsc: false,
+  closeOnOutsideClick: false,
+  closeOnResize: false,
+  contentHeightMax: -1,
+  contentWidthMax: -1,
+  isHeaderDisabled: false,
+};
+
+class PopupContent extends React.Component {
+  static getContentStyle(arrow, arrowPosition, height, maxHeight, width, maxWidth) {
+    const validHeight = maxHeight <= 0 || height <= maxHeight ? height : maxHeight;
+    const validWidth = maxWidth <= 0 || width <= maxWidth ? width : maxWidth;
+    return { height: `${validHeight.toString()}px`, width: `${validWidth.toString()}px` };
+  }
+
+  static addPopupHeader(children, onRequestClose) {
+    const icon = <IconClose height="20" width="20" />;
```

Comment:
Talking with Neil, for now we are going to go with the px value instead of scaling, but that may change down the road as the design for the modal and detail view dialogs are firmed up.